chef/cookbooks/bind9/recipes/default.rb

```diff
@@ -277,8 +311,21 @@ def make_zone(zone)
 end
 
 # Write out the zone databases that Crowbar will be responsible for.
+entries = []
 zones.keys.sort.each do |zone|
-  make_zone zones[zone]
+  entries << make_zone(zones[zone])
+end
+
+# Drop files that are not used anymore (forgotten nodes, deallocated IP, etc.)
+entries.flatten!
+db_files = Dir.glob("/etc/bind/db.*")
+db_files.each do |db_file|
+  entry = db_file["/etc/bind/db.".length..-1]
+  unless entries.include? entry
```

Comment:
Style/Next: Use `next` to skip iteration. (https://github.com/bbatsov/ruby-style-guide#no-nested-conditionals)